File: DotNetGotchas/VBDN/Delegate/UnInitializedDelegate/AComponent.vb
'AComponent.vb
Public Delegate Sub DummyDelegate()

Public Class AComponent
    Public Event myEvent As DummyDelegate

    Protected Overridable Sub OnMyEvent()
        RaiseEvent myEvent()
    End Sub

    Public Sub fire()
        Console.WriteLine("Raising event")
        OnMyEvent() ' Raising the event
        Console.WriteLine("Done raising event")
    End Sub
End Class

File: DotNetGotchas/CSharp/Delegate/UnInitializedDelegate/AComponent.cs
// AComponent.cs
using System;

namespace UnInitializedDelegate
{
	public delegate void DummyDelegate();

	public class AComponent
	{
		public event DummyDelegate myEvent;

		protected virtual void OnMyEvent()
		{
			myEvent();
		}

		public void Fire()
		{
			Console.WriteLine("Raising event");
			OnMyEvent(); // Raising the event
			Console.WriteLine("Done raising event");
		}
	}
}
